app/api/pathway/search/route.ts

```diff
@@ -17,60 +16,97 @@ export async function GET(request: NextRequest) {
     )
   );
 
-  let blob = pathways;
-  let flatten: any = {};
-  for (var [name, v] of Object.entries(blob)) {
-    let coursesIn = [];
-    let department = "No Data";
-    for (var [k, c] of Object.entries(v)) {
-      if (typeof c === "object" && k != "minor") {
-        for (var [title, code] of Object.entries(c)) {
-          coursesIn.push(title);
-        }
+  let pathwayResult: IPathwaySchema[] = [];
+
+  for (var [pathwayName, data] of Object.entries(pathways)) {
+    let description: string = "";
+    let remainingHeader: string = "";
+    let schema: Array<ICourseClusterSchema> = [];
+    let allCourses: Array<string> = [];
+    let department: string = "No Data";
+    let resultPathway: IPathwaySchema = {
+      title: pathwayName,
+      description: "no pathway found",
+      compatibleMinor: ["no pathway found"],
+      department: department,
+      coursesIn: ["no pathway found"],
+      clusters: []
+    }
+    for (let key of Object.keys(pathways[pathwayName])) {
+      if (key == "description"){
+          description = pathways[pathwayName][key];
+      }
+      else if (key == "remaining_header"){
+          remainingHeader = pathways[pathwayName][key];
+      }
+      else if (key == "name" || key == "minor"){
+          continue;
+      }
+      else{
+          let temp_courses: Array<string> = [];
+          for (let course_name of Object.keys(pathways[pathwayName][key])){
+              temp_courses.push(course_name);
+              allCourses.push(course_name);
+          }
+          let temp_description: string = "";
+          let temp_name: string = key;
+          if (key == 'Remaining'){
+              temp_description = pathways[pathwayName]["remaining_header"] ? pathways[pathwayName]["remaining_header"] : "Take your remaining courses from this list:";
+          } else{
+              if (key == "Required"){
+                  if (temp_courses.length == 1){
+                      temp_description = "You must take the following course:";
+                  } else {
+                      temp_description = "Take all of the following courses:";
+                  }
+              } else {
+                  temp_description = "You must take one of the following courses:";
+                  temp_name = "One Of";
+              }
+          }
+          let num = 3;
+          if (key != "Remaining"){
+              num = 1;
+          }
+          if (key == "Required"){
+              num = temp_courses.length;
+          }
+          let cluster: ICourseClusterSchema = {
+              name: temp_name,
+              numCourses: num,
+              description: temp_description,
+              courses: temp_courses
+          };
+          schema.push(cluster);
```

Comment:
The reason I did that was because there are a few different cluster names. I can do it manually, but I wanted to do it this way so that I didn't have to check for all of the possible cluster names from previous jsons,